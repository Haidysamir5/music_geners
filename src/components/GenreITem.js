import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Modal, Button } from "antd";
import GenreArtists from "./GenreArtists";
class GenreItem extends Component {

  state = { visible: false }; //modal state

  showModal = (id) => {
    this.setState({
      visible: true,
    });
    window.history.replaceState(null, null, `${id}`); //change url to add id without reload 
  };

  handleCancel = (e) => {
    this.setState({
      visible: false,
    });
    this.handleClose();
  };
  handleClose = (e) => {

    window.history.replaceState({}, document.title, "/" + ""); //change url to remove id without reload 

  };

  render() {
    const { props } = this.props;
    let { id , name, picture_big, type } = props;
    return (
      <div className="col-12 col-md-4 col-lg-3">
        <div className="single-show-item mb-30">
          <img src={picture_big} alt="" />
          <div className="overlay-content">
            <div className="text-center">
              <span>{type}</span>

             <Link to={`/${id}`} target="_blank" className="link"><h5>{name}</h5></Link> 
             
              <Button
                className="mt-2 btn  btn-pink "
                type="primary"
                onClick={() => {
                  this.showModal(id);
                }}
              >
                Show artists
              </Button>
              
             
            </div>
          </div>
          <Modal
                style={{ top: 20 }}
                width={900}
                footer={null}
                title="Artists"
                visible={this.state.visible}
                onCancel={this.handleCancel}
                onClose={this.handleClose}
              >
                {<GenreArtists id={id} />}
              </Modal>
        </div>
      </div>
    );
  }
}

export default connect()(GenreItem);

