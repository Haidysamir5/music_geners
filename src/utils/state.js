export const music={
    modal_state:false,
    music_info:[]
}