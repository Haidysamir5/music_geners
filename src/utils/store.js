import { createStore, applyMiddleware } from "redux";
import promiseMiddleware from "redux-promise";
import reducers from "../reducers";
export const store = createStore(reducers, applyMiddleware(promiseMiddleware));

