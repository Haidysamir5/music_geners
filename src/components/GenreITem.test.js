import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import GenreITem from "./GenreITem";
// wrap the root component in a Provider
import { store } from "../utils/store";
import { Modal, Button } from "antd";

test("match snapshoot", () => {
  const wrapper = shallow(<GenreITem store={store} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

describe("Test Button component", () => {
  it("Test click event", () => {
    const showModal = jest.fn();

    const button = shallow(
      <Button
        className="mt-2 btn  btn-pink "
        type="primary"
        onClick={showModal}
      >
        Show artists
      </Button>
    );
    button.find("button").simulate("click");
    expect(showModal.mock.calls.length).toEqual(1);
  });

});
