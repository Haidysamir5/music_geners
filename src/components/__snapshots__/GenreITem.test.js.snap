// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshoot 1`] = `
<GenreItem
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
